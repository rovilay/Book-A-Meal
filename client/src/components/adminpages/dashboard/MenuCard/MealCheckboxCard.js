/* eslint react/no-unused-prop-types:0 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ReactPaginate from 'react-paginate';

class MealCheckBoxCard extends Component {
  constructor(props) {
    super(props);

    this.handlePaginationClick = this.handlePaginationClick.bind(this);
  }


  handlePaginationClick(data) {
    const { getMeals, mealPagination } = this.props;
    const { limit } = mealPagination;
    const currentPage = data.selected;
    const nextOffset = currentPage * limit;

    getMeals({ limit, offset: nextOffset });
  }

  render() {
    const {
      newMenuMeals,
      mealPagination,
      setNewMenuMeal
    } = this.props;

    const {
      numOfPages,
      count
    } = mealPagination;

    return (
      <div className="mealsCheckbox">
        <div className="checkbox-card">
          { this.props.meals.map(meal => (
            <p key={meal.id}>
              <label htmlFor={meal.id}>
                {meal.title}
              </label>
              <input
                type="checkbox"
                name="meal-check"
                className="meal-check"
                id={`checkbox-${meal.id}`}
                defaultChecked={newMenuMeals.includes(meal.id)}
                onClick={() => {
                  setNewMenuMeal(meal.id);
                }
                }
                value={meal.id}
              />
            </p>
          ))}
        </div>
        {
          (count > 10)
          &&
          <div className="pagination-container">
            <ReactPaginate
              previousLabel="<<"
              nextLabel=">>"
              breakLabel={<a href="">...</a>}
              breakClassName="break-me"
              pageCount={numOfPages}
              marginPagesDisplayed={2}
              pageRangeDisplayed={5}
              onPageChange={this.handlePaginationClick}
              containerClassName="pagination"
              subContainerClassName="pages pagination"
              activeClassName="active"
            />
          </div>
        }
      </div>
    );
  }
}

MealCheckBoxCard.propTypes = {
  meals: PropTypes.array.isRequired,
  setNewMenuMeal: PropTypes.func.isRequired,
  mealPagination: PropTypes.object.isRequired,
  getMeals: PropTypes.func.isRequired,
  newMenuMeals: PropTypes.array.isRequired
};

export default MealCheckBoxCard;
