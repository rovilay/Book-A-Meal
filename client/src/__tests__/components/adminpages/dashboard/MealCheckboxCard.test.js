import React from 'react';
import { shallow } from 'enzyme';
import MealCheckboxCard from '../../../../components/adminpages/dashboard/MenuCard/MealCheckboxCard';
import { menuMeals } from '../../../__mockData__/menuMock';


const props = {
  meals: menuMeals,
  setNewMenuMeal: jest.fn(),
  mealPagination: {
    limit: 12,
    count: 13,
    numOfPages: 2
  },
  getMeals: jest.fn(),
  newMenuMeals: menuMeals.slice(0, 2)
};

describe('MealCheckboxCard component test', () => {
  const setup = () => shallow(<MealCheckboxCard {...props} />);


  it('should render MealCheckboxCard correctly', (done) => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();

    done();
  });


  it('should call `handlePaginationClick` meal pagination button is clicked',
  (done) => {
    const wrapper = setup();

    const data = {
      selected: 1
    };

    const handlePaginationClickSpy= jest.spyOn(
      wrapper.instance(), 'handlePaginationClick'
    );
    wrapper.instance().handlePaginationClick(data)
    expect(handlePaginationClickSpy).toHaveBeenCalled();

    done();
  });

  it('should call `setNewMenuMeal` if meals checked box is checked', (done) => {
    const wrapper = setup();

    wrapper.find(`.meal-check`).first().simulate('click');
    expect(props.setNewMenuMeal).toHaveBeenCalled()

    done();
  });
});
