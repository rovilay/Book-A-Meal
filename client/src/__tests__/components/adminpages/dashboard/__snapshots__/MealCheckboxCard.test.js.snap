// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealCheckboxCard component test should render MealCheckboxCard correctly 1`] = `
<div
  className="mealsCheckbox"
>
  <div
    className="checkbox-card"
  >
    <p
      key="1"
    >
      <label
        htmlFor="1"
      >
        EBA AND EGUSI
      </label>
      <input
        className="meal-check"
        defaultChecked={false}
        id="checkbox-1"
        name="meal-check"
        onClick={[Function]}
        type="checkbox"
        value="1"
      />
    </p>
    <p
      key="2"
    >
      <label
        htmlFor="2"
      >
        SALAD
      </label>
      <input
        className="meal-check"
        defaultChecked={false}
        id="checkbox-2"
        name="meal-check"
        onClick={[Function]}
        type="checkbox"
        value="2"
      />
    </p>
    <p
      key="3"
    >
      <label
        htmlFor="3"
      >
        SPAGHETTI
      </label>
      <input
        className="meal-check"
        defaultChecked={false}
        id="checkbox-3"
        name="meal-check"
        onClick={[Function]}
        type="checkbox"
        value="3"
      />
    </p>
    <p
      key="4"
    >
      <label
        htmlFor="4"
      >
        YAM AND EGG
      </label>
      <input
        className="meal-check"
        defaultChecked={false}
        id="checkbox-4"
        name="meal-check"
        onClick={[Function]}
        type="checkbox"
        value="4"
      />
    </p>
  </div>
  <div
    className="pagination-container"
  >
    <PaginationBoxView
      activeClassName="active"
      breakClassName="break-me"
      breakLabel={
        <a
          href=""
        >
          ...
        </a>
      }
      containerClassName="pagination"
      disableInitialCallback={false}
      disabledClassName="disabled"
      marginPagesDisplayed={2}
      nextClassName="next"
      nextLabel=">>"
      onPageChange={[Function]}
      pageCount={2}
      pageRangeDisplayed={5}
      previousClassName="previous"
      previousLabel="<<"
      subContainerClassName="pages pagination"
    />
  </div>
</div>
`;
